### PR TITLE
fix: add mains to samples

### LINKED ISSUE #2873: dataproc.quickstart.quickstart_test: test_quickstart failed
dataproc.quickstart.quickstart_test: test_quickstart failed
buildID: fbcdef163f965e363c6b1abf9d41b14fde28271e
buildURL: https://source.cloud.google.com/results/invocations/9b3b09c2-b687-42e2-9066-613031c2b1ec
status: failed

### PR DESCRIPTION
**Pull request by bradmiro**
Added mains to two samples: create_cluster and instantiate_inline_workflow_templates.

Fixed their associated tests to accommodate this.

Removed subprocess from quickstart/quickstart_test.py to fix [2873](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2873)

fixes #2873

### CODE REVIEW

File: dataproc/instantiate_inline_workflow_template_test.py [hunk 1]
```diff
@@ -13,19 +13,21 @@
...
-    instantiate_inline_workflow_template.instantiate_inline_workflow_template(
-        PROJECT_ID, REGION
-    )
+    command = [
+        'python', 'instantiate_inline_workflow_template.py',
+        PROJECT_ID,
+        REGION,
+    ]
+
+    out = subprocess.check_output(command).decode("utf-8")
```
Comment (tmatsuo) [inline dataproc/instantiate_inline_workflow_template_test.py:31]: Please don't spawn a process unless it's extremely necessary.
Options:
1) just keep the current test
2) have`main(argv=None)` and test `main(my_argv)`

```python
def main(argv=sys.argv):
    # use argv in your code....

if __name__ == "__main__":
    main()
```

**Comment by tmatsuo** [comment]
@bradmiro 
nit: it seems like we use present tense for commit messages, so maybe `fix: add mains to samples`?

File: dataproc/create_cluster.py [hunk 2]
```diff
@@ -58,4 +64,11 @@ def create_cluster(project_id, region, cluster_name):
...
     # Output a success message.
     print('Cluster created successfully: {}'.format(result.cluster_name))
-    # [END dataproc_create_cluster]
+
+
+if __name__ == "__main__":
+    project_id = sys.argv[1]
+    region = sys.argv[2]
+    cluster_name = sys.argv[3]
+    create_cluster(project_id, region, cluster_name)
```
[the remark this answers is not in the document]
Comment (bradmiro) [reply dataproc/create_cluster.py]: Done.

File: dataproc/instantiate_inline_workflow_template.py [hunk 3]
```diff
@@ -91,8 +92,10 @@ def instantiate_inline_workflow_template(project_id, region):
 
     # Output a success message.
     print('Workflow ran successfully.')
-# [END dataproc_instantiate_inline_workflow_template]
 
 
 if __name__ == "__main__":
-    instantiate_inline_workflow_template(sys.argv[1], sys.argv[2])
+    project_id = sys.argv[1]
```
[the remark this answers is not in the document]
Comment (bradmiro) [reply dataproc/instantiate_inline_workflow_template.py]: Done.

File: dataproc/instantiate_inline_workflow_template_test.py (hunk 1 above)
Comment (bradmiro) [reply dataproc/instantiate_inline_workflow_template_test.py]: Done. Kept original tests.

**Comment by bradmiro** [comment]
Thanks for the feedback Takashi.


> @bradmiro
> nit: it seems like we use present tense for commit messages, so maybe `fix: add mains to samples`?

Fi

**Comment by bradmiro** [comment]
Didn't mean to close, please ignore that.

File: dataproc/create_cluster.py [hunk 4]
```diff
@@ -33,7 +39,7 @@ def create_cluster(project_id, region, cluster_name):
 
     # Create a client with the endpoint set to the desired cluster region.
     cluster_client = dataproc.ClusterControllerClient(client_options={
-        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(region)
+        'api_endpoint': '{}-dataproc.googleapis.com:443'.format(region),
```
Comment (tmatsuo) [inline dataproc/create_cluster.py:42]: Do you need to add the comma?

File: dataproc/create_cluster.py [hunk 5]
```diff
@@ -58,4 +64,15 @@ def create_cluster(project_id, region, cluster_name):
 
     # Output a success message.
     print('Cluster created successfully: {}'.format(result.cluster_name))
-    # [END dataproc_create_cluster]
+
+
+if __name__ == "__main__":
+    if len(sys.arv) != 4:
```
Comment (tmatsuo) [inline dataproc/create_cluster.py:70]: `sys.argv` instead of `sys.arv`

File: dataproc/create_cluster.py [hunk 6]
```diff
@@ -58,4 +64,15 @@ def create_cluster(project_id, region, cluster_name):
...
+        print("INSUFFICIENT ARGS: Please provide a "
```
Comment (tmatsuo) [inline dataproc/create_cluster.py:71]: How about:

```python
if len(sys.argv) < 4:
    sys.exit('python create_cluster.py project_id region cluster_name')
```
?

File: dataproc/instantiate_inline_workflow_template.py [hunk 7]
```diff
@@ -91,8 +92,13 @@ def instantiate_inline_workflow_template(project_id, region):
 
     # Output a success message.
     print('Workflow ran successfully.')
-# [END dataproc_instantiate_inline_workflow_template]
 
 
 if __name__ == "__main__":
-    instantiate_inline_workflow_template(sys.argv[1], sys.argv[2])
+    if len(sys.arv) != 3:
```
Comment (tmatsuo) [inline dataproc/instantiate_inline_workflow_template.py:98]: ditto

File: dataproc/instantiate_inline_workflow_template.py [hunk 8]
```diff
@@ -91,8 +92,13 @@ def instantiate_inline_workflow_template(project_id, region):
...
+        print("INSUFFICIENT ARGS: Please provide a PROJECT_ID and REGION.")
```
Comment (tmatsuo) [inline dataproc/instantiate_inline_workflow_template.py:99]: ditto

File: dataproc/create_cluster.py (hunk 4 above)
Comment (bradmiro) [reply dataproc/create_cluster.py]: Fixed.

File: dataproc/create_cluster.py (hunk 5 above)
Comment (bradmiro) [reply dataproc/create_cluster.py]: Fixed.

File: dataproc/create_cluster.py (hunk 6 above)
Comment (bradmiro) [reply dataproc/create_cluster.py]: Fixed.

File: dataproc/instantiate_inline_workflow_template.py (hunk 7 above)
Comment (bradmiro) [reply dataproc/instantiate_inline_workflow_template.py]: Fixed.

File: dataproc/instantiate_inline_workflow_template.py (hunk 8 above)
Comment (bradmiro) [reply dataproc/instantiate_inline_workflow_template.py]: Fixed.

**Comment by bradmiro** [comment]
Thanks for the review. Ack that your `main(argv=sys.argv)` solution would work here but I do not think it would come across as clearly in the sample.

**Comment by tmatsuo** [comment]
> Thanks for the review. Ack that your `main(argv=sys.argv)` solution would work here but I do not think it would come across as clearly in the sample.

Well, do you really need to `if __name__ == '__main__':` part in the code snippet? If so I'm fine with the current state, but as a reader, I'd think having the function itself is better.

If I were you, I would only have the function itself in the code snippet, have the `main(argv=sys.argv)` and test it.

But it's up to you.

File: dataproc/create_cluster.py (hunk 6 above)
Comment (tmatsuo) [reply dataproc/create_cluster.py:71]: The messages seems reverted to the previous state?

Comment (bradmiro) [reply dataproc/create_cluster.py]: Fixed again.

**Comment by bradmiro** [comment]
Removed subprocess from test in quickstart/quickstart_test.py to fix flaky test per issue [2873](https://github.com/GoogleCloudPlatform/python-docs-samples/issues/2873)

File: dataproc/create_cluster.py (hunk 4 above)
Comment (leahecole) [inline dataproc/create_cluster.py:42]: nit should we be using f-strings instead of `.format` to be more modern?

File: dataproc/create_cluster.py [hunk 9]
```diff
@@ -59,3 +65,13 @@ def create_cluster(project_id, region, cluster_name):
     # Output a success message.
     print('Cluster created successfully: {}'.format(result.cluster_name))
```
Comment (leahecole) [inline dataproc/create_cluster.py:66]: another potential f-string spot

**Pull request merged**